pkg/s3select/helpers.go

```diff
@@ -193,6 +190,138 @@ func applyStrFunc(rawArg string, funcName string) string {
 
 }
 
+// evaluateBetween is a function which evaluates a Between Clause.
+func evaluateBetween(betweenExpr *sqlparser.RangeCond, alias string, record string) (bool, error) {
+	var colToVal interface{}
+	var colFromVal interface{}
+	var conversionColumn string
+	var funcName string
+	switch colTo := betweenExpr.To.(type) {
+	case sqlparser.Expr:
+		switch colToMyVal := colTo.(type) {
+		case *sqlparser.FuncExpr:
+			var temp string
+			temp = stringOps(colToMyVal, record, "")
```

Comment:
why is this converted to a byte array here and not in the following piece of code?